File: tests/aws/services/lambda_/functions/common/introspection/java/src/src/main/java/echo/Handler.java
package echo;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.RequestHandler;


import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class ReturnValue {
    public Map<String, String> environment;
    public Map<String, String> ctx;
    public List<String> packages;

    public ReturnValue(Context context) {
        this.environment = System.getenv();
        this.ctx = new HashMap<>();
        this.ctx.put("function_name", context.getFunctionName());
        this.ctx.put("function_version", context.getFunctionVersion());
        this.ctx.put("invoked_function_arn", context.getInvokedFunctionArn());
        this.ctx.put("memory_limit_in_mb", Integer.toString(context.getMemoryLimitInMB()));
        this.ctx.put("aws_request_id", context.getAwsRequestId());
        this.ctx.put("log_group_name", context.getLogGroupName());
        this.ctx.put("log_stream_name", context.getLogStreamName());
        this.ctx.put("remaining_time_in_millis", Integer.toString(context.getRemainingTimeInMillis()));
        this.packages = new ArrayList<>();
    }
}

public class Handler implements RequestHandler<Map<String, String>, ReturnValue> {

    public ReturnValue handleRequest(Map<String, String> event, Context context) {
        return new ReturnValue(context);
    }
}

File: tests/aws/services/lambda_/functions/common/introspection/python/src/handler.py
import json
import os


def handler(event, context):
    print(json.dumps(event))
    return {
        "environment": dict(os.environ),
        "ctx": {
            "function_name": context.function_name,
            "function_version": context.function_version,
            "invoked_function_arn": context.invoked_function_arn,
            "memory_limit_in_mb": context.memory_limit_in_mb,
            "aws_request_id": context.aws_request_id,
            "log_group_name": context.log_group_name,
            "log_stream_name": context.log_stream_name,
            "remaining_time_in_millis": context.get_remaining_time_in_millis(),
        },
        "packages": [],
    }
